components/aws/sagemaker/tests/integration_tests/component_tests/test_batch_transform_component.py

```diff
@@ -0,0 +1,78 @@
+import utils
+import os
+import pytest
+
+from utils import kfp_client_utils
+from utils import minio_utils
+from utils import sagemaker_utils
+from utils import s3_utils
+
+
+@pytest.mark.parametrize(
+    "test_file_dir", ["resources/config/kmeans-mnist-batch-transform"]
+)
+def test_transform_job(
+    kfp_client,
+    experiment_id,
+    s3_client,
+    sagemaker_client,
+    s3_data_bucket,
+    test_file_dir,
+):
+
+    download_dir = utils.mkdir(os.path.join(test_file_dir + "/generated"))
+    test_params = utils.load_params(
+        utils.replace_placeholders(
+            os.path.join(test_file_dir, "config.yaml"),
+            os.path.join(download_dir, "config.yaml"),
+        )
+    )
+
+    # Generate random prefix for model, job name to avoid errors if resources with same name exists
+    test_params["Arguments"]["model_name"] = test_params["Arguments"][
+        "job_name"
+    ] = input_job_name = (
+        utils.generate_random_string(5) + "-" + test_params["Arguments"]["model_name"]
```

Comment:
Should this be timestamped instead?

Reply:
I thought the same but couldn't think of any advantage since we should be deleting the resources anyway.

Reply:
If we can tie the model back to this particular test run, that would be helpful for debugging.

Reply:
nope because it creates problems when tests are run in parallel. All console will have filters to sort according to time and in addition to that logs will help pin point which model/config to look at.

Note we can now run tests in parallel by using `pytest -n <NUM_OF_PARALLEL_TESTS>`